android/jni/com/mapswithme/core/jni_helper.cpp

```diff
@@ -44,8 +44,10 @@ extern "C"
 //    g_findClassMethod = env->GetMethodID(classLoaderClass, "findClass",
 //                                    "(Ljava/lang/String;)Ljava/lang/Class;");
 //    ASSERT(g_findClassMethod, ("FindClass methodId can't be 0"));
-    g_indexClazz = static_cast<jclass>(env->NewGlobalRef(env->FindClass("com/mapswithme/maps/MapStorage$Index")));
-    ASSERT(g_indexClazz, ("Index class not found!"));
+
+    g_indexClazz = jni::GetGlobalClassRef(env, "com/mapswithme/maps/MapStorage$Index");
+    g_mapObjectClazz = jni::GetGlobalClassRef(env, "com/mapswithme/maps/bookmarks/data/MapObject");
```

Comment:
Разве все классы в андроиде не выгружаются, когда останавливается JVM? Интересно, за цикл жизни нашей JNI части в JVM это происходит или нет?

Думаю это не критично, для scoped guard все равно надо было бы явно вызывать деструкторы в OnUnload().

Для информации:
```
JNI_OnUnload
void JNI_OnUnload(JavaVM *vm, void *reserved);

The VM calls JNI_OnUnload when the class loader containing the native library is garbage collected. This function can be used to perform cleanup operations. Because this function is called in an unknown context (such as from a finalizer), the programmer should be conservative on using Java VM services, and refrain from arbitrary Java call-backs.
Note that JNI_OnLoad and JNI_OnUnload are two functions optionally supplied by JNI libraries, not exported from the VM.
```